### ISSUE TITLE
Proguard File

### ISSUE THREAD

**Issue by ntoper** [duckduckgo/Android#444]
Hi,

I think deploying proguard efficiently would reduce the size of the app by 30%. I'd love to do that for you. The contribution guide mentions I should log an issue first before starting to work on it. Let me know

**Comment by subsymbolic** [comment]
This is an exciting suggestion. We’ve really interested in the potential performance improvements, less so apk shrinkage.  If you can achieve a measurable performance gain, particularly a reduction in page load times, we would add it. If not, we would avoid it for now as it comes with a long term maintenance and testing overhead.

One question, would you consider R8 over ProGuard?

**Comment by ntoper** [comment]
I profiled the app for performance. Issues I have seen so far are coming from: 

 com.duckduckgo.app.entities.entityForUrl(url: String): EntityListEntity (>3s to run in my profiler) This looks easy to test and would speed up the codebase tremendously. Should I do it? If yes, I'll create another issue

R8 is better than Proguard btw yes

**Comment by subsymbolic** [comment]
@ntoper 3s is huge, great catch. Yes, feel free to create an issue to fix this.

**Comment by subsymbolic** [comment]
@ntoper we're most keen on the ProGuard / R8 optimizations so if you want to spend time on that please do. Regarding the performance issue you found, we were going to fix that pretty soon, but if you wanted to look into it you may be able to sneak in a PR before we get to it. Either way, thanks again for letting us know about it.

**Comment by ntoper** [comment]
- Proguard/R8 optimization won't help imho but I will give it a try to confirm. I already have a working file with a 30% size reduction
- do you need an issue on the bottleneck or you're good to go? I can fix it, but if you will address it already it's better you do as you know this part better (and it's used everywhere)

**Comment by subsymbolic** [comment]
I noticed you logged the bottleneck for us in https://github.com/duckduckgo/Android/issues/453. That's perfect, thanks for doing that.

**Comment by subsymbolic** [comment]
Closing as we have no current plans to reduce the apk file size.
